### DISCUSSION TITLE
File changes watching only on specified files

### DISCUSSION THREAD

**Discussion by lordanubi** [denoland/deno#14912] [Q&A]
Hi is there any way to make the server restart only if you edit certain files? I have a main.jsx file I'd like to watch for changes, this file imports my App.jsx file which is server side rendered. I want changes done on main.jsx to restart the server but not the changes on App.jsx (I already have an HMR feature for that).
I tried this
`deno run --watch=Server/main.jsx Server/main.jsx`
but editing App.jsx still affects the server restarting, I think it's because I'm importing that file in main.jsx like so:
`import App from './index.jsx'
`
I also tried to move it out of the main.jsx directory but is still being watched

**Comment by bartlomieju** [comment]
No, it is not possible - all modules that are loaded into the runtime (either by `import` statement or dynamic `import()` call) are being watched for changes. Fresh is using a similar approach and that doesn't seem to be a problem. @lucacasonato thoughts?

**Comment by lordanubi** [comment]
Well yeah actually is not that bad cause I have instant change with HMR.
It's just that after a little bit I also get a page refresh and it's a bit annoying for performance but if there's no way to prevent that I'll leave it like this :)

**Comment by finalclass** [comment]


**Comment by bebraw** [comment]
It might be bit of a kludge but perhaps one option would be to implement your own wrapper using [Deno.watchFs](https://deno.land/api@v1.38.5?s=Deno.watchFs). At least then you would have more control over the behavior with the cost of having more code of yours to maintain.
